components/Feed/Caption.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+import { useEffect, useState } from "react";
+
+export default function Caption({ caption }: { caption: string | null }) {
+  // track if the caption is truncated
+  const [isCaptionTruncated, setIsCaptionTruncated] = useState<boolean>(false);
+  // caption shown on the video slide
+  const [displayCaption, setDisplayCaption] = useState<string | null>(caption);
+  // track if the caption is expanded
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const truncationCharacterLimit = 80;
+
+  // truncate the caption if it's is greater than truncationCharacterLimit
+  useEffect(() => {
+    if (caption && caption.length > 100) {
+      setIsCaptionTruncated(true);
+      setDisplayCaption(caption.slice(0, truncationCharacterLimit) + "...");
+    }
+  }, [caption]);
+
+  // toggle expansion state when the "more" or "less" button is clicked
+  const handleExpansion = () => {
+    setIsExpanded(!isExpanded);
+    if (isExpanded) {
+      setDisplayCaption(caption?.slice(0, truncationCharacterLimit) + "...");
+    } else {
+      setDisplayCaption(caption);
+    }
+  };
+
+  return (
+    <div>
+      <p className="inline">{displayCaption}</p>
```

Comment:
What made you choose to use a `<p className="inline">` instead of just wrapping it all in a single `<p>` div?

Reply:
The "more" or "less" button moved to the next line in the original code but it appears to work correctly now so I removed the inline.